### PR TITLE
One minor one major fix

### PR DESCRIPTION
**Pull request by lfrancke**
Minor: This allows using the `object_to_owner_reference` method with more objects.
`Meta` requires `Resource` but we only need `Resource`.

Major: This also fixes ConfigMap creation to use the namespace of the resource that's passed in.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -98,7 +99,7 @@ pub fn create_tolerations() -> Vec<Toleration> {
     ]
 }
 
-pub fn object_to_owner_reference<K: Meta>(meta: ObjectMeta) -> OperatorResult<OwnerReference> {
+pub fn object_to_owner_reference<K: Resource>(meta: ObjectMeta) -> OperatorResult<OwnerReference> {
```
Comment (soenkeliebau) [inline src/lib.rs:102]: I won't pretend to fully understand the difference, but could you just briefly comment on why the other Metas don't need to be substituted by "Resource"?

**Review by soenkeliebau — COMMENTED** [review]
Left one quick question

File: src/lib.rs (hunk 1 above)
Comment (lfrancke) [reply src/lib.rs:102]: The only thing we access from `K` are the associated consts `API_VERSION` and `KIND` and they are from `Resource`.

`Resource` is a supertrait for `Meta` so this works as well but we don't need to be as _tight_.
Every `Meta` is also a `Resource` but not every `Resource` does have to be `Meta`.

**Review by soenkeliebau — APPROVED** [review]
LGTM

**Pull request merged**